src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataCollectionSerializer.cs

```diff
@@ -161,17 +174,25 @@ public ODataCollectionSerializer(ODataSerializerProvider serializerProvider)
                     IEdmTypeReference actualType = writeContext.GetEdmType(item, item.GetType());
                     Contract.Assert(actualType != null);
 
-                    itemSerializer = itemSerializer ?? SerializerProvider.GetEdmTypeSerializer(actualType);
-                    if (itemSerializer == null)
+                    if (isForAnnotations && actualType.IsComplex())
```

Comment:
>isForAnnotations [](start = 24, length = 16)

we should have a comment to describe why we need this (somewhat ugly) special case.

Reply:
#Resolved